WordPress/gutenberg
# Post fields: move `comment_status` from edit-site to fields package

Part of https://github.com/WordPress/gutenberg/issues/61084

## What?

This PR moves the `comment_status` field definition from `edit-site` to the `fields` package.

## Why?

In preparation for moving the whole `usePostFields` hook ([conversation](https://github.com/WordPress/gutenberg/pull/65196#discussion_r1777506405)).

## How?

- 36f4681e387ac84491bcd367ec44686818a1ad75

## Testing Instructions

Visit the Pages page and verify that the "Discussion" field still works as expected.

<img width="1533" alt="Screenshot 2024-11-12 at 16 10 17" src="https://github.com/user-attachments/assets/603f3c4e-c5d2-44ce-9862-53792f8bb636">
